Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -1257,24 +1260,32 @@ namespace AzToolsFramework
                 return AZStd::move(retrieveEntitiesAndInstancesOutcome);
             }
 
+            // Take a snapshot of the instance DOM before we manipulate it.
+            Prefab::PrefabDom instanceDomBefore;
+            m_instanceToTemplateInterface->GenerateDomForInstance(instanceDomBefore, commonOwningInstance->get());
+
             for (auto& nestedInstance : instances)
             {
+                AZ::EntityId containerEntityId = nestedInstance->GetContainerEntityId();
                 AZStd::unique_ptr<Instance> outInstance =
                     commonOwningInstance->get().DetachNestedInstance(nestedInstance->GetInstanceAlias());
                 RemoveLink(outInstance, commonOwningInstance->get().GetTemplateId(), undoBatch.GetUndoBatch());
                 outInstance.reset();
-            }
 
-            m_instanceToTemplateInterface->GenerateDomForInstance(instanceDomBefore, commonOwningInstance->get());
+                m_prefabUndoCache.PurgeCache(containerEntityId);
+            }
 
             for (AZ::Entity* entity : entities)
             {
+                AZ::EntityId entityId = entity->GetId();
                 commonOwningInstance->get().DetachEntity(entity->GetId()).release();
-                AZ::ComponentApplicationBus::Broadcast(&AZ::ComponentApplicationRequests::DeleteEntity, entity->GetId());
+                AZ::ComponentApplicationBus::Broadcast(&AZ::ComponentApplicationRequests::DeleteEntity, entityId);
+                
+                m_prefabUndoCache.PurgeCache(entityId);
```

Comment:
why is this needed here? When entities get destructed, I would assume that they are getting removed from the cache. Can you check if that's the case? Steps are: remove this line, Delete an entity, wait for propagation to finish and look up the undo cache later to see whether the id is still present.

Reply:
That's right. I verified the entity cache gets wiped out.